packages/desktop/views/dashboard/wallet/views/multi-step/MultiStep.svelte

```diff
@@ -0,0 +1,32 @@
+<script lang="ts">
+    import { ImplicitAccountCreationState } from '@contexts/implicit-account-creation'
+
+    const IMPLICIT_ACCOUNT_STEPS: ImplicitAccountCreationState[] = Object.values(ImplicitAccountCreationState)
+
+    let activeState: ImplicitAccountCreationState | null = null
+    let onNext: () => Promise<boolean>
+    let currentStep: number = 0
+    let activeComponent
+
+    async function loadComponent() {
+        activeState = IMPLICIT_ACCOUNT_STEPS[currentStep]
+        const component = await import(`./${activeState}.svelte`)
+        activeComponent = component.default
+    }
+
+    async function handleNext() {
+        currentStep += 1
+        await loadComponent()
+    }
+
+    $: loadComponent()
```

Comment:
careful with these, they are not really reactive so its better putting them inside an `onMount` if that is what you are looking for